test/numbers.coffee

```diff
@@ -86,3 +86,10 @@ test "Infinity", ->
 
 test "NaN", ->
   ok isNaN 1/NaN
+
+test "#3828: Floats with leading periods still work", ->
+  eq (0.5).toFixed(1), (.5).toFixed(1)
+
+  foo = (n) -> n
+  eq 0.2, foo .2
```

Comment:
This really _shouldn’t_ work . . . in JavaScript, there’s a good chance that `0.2 === 0.2` fails, because one of them might be treated as `0.20000000001` or so.

Reply:
No, that’s not true.

Reply:
I think this one can fail sometimes `.2 === 2/10 === .1 * 2`.

Reply:
The literals would be fine - it would be a different story if it was testing...

```coffee
foo = (n) -> n + 0.2
eq 0.6, foo .4
```

...because `0.2 + 0.4 isnt 0.6` (it's `0.6000000000000001`).